#!/usr/bin/env Rscript

###### 
# This script will output a table with the number of sequences generated per sample 
# for Gorka's data
# input: 	file with lines in each fastq file
# output: 	table with reads sequenced per sample
#			figure with the number of reads sequenced per sample
######


###### PARAMETERS ##########
# Set the parameters:
today <- Sys.Date()											# Set the date that will go on the end of the files generated by this script
today <- format(today, format="%m%d%y")
#############################



##### Load in the reads sequenced, after read QC, and after classification QC per sample:
seqed <- read.table(file="../results/table.reads.sequenced.per.sample.060614ERD.txt", sep="\t", header=TRUE)
qced <- read.table(file="../results/table.reads.sequenced.per.sample.after.QC.060614ERD.txt", sep="\t", header=TRUE)
classified <- read.table(file="/Users/erdavenport/clusterhome/gorka/data/summaries/lines_after_classification_QC_060914ERD.txt", sep=" ", header=FALSE)

classified$V2 <- gsub(".R1_R2_single100F_rdp_genus_nr_rdp_ltp_rearranged_genus_trimmed.taxonomy.CI.FILTERED.taxonomy", "", classified$V2)
write.table(classified, paste("../results/table.reads.classified.per.sample.",today,"ERD.txt", sep=""), row.names=FALSE, quote=FALSE)

seqed$V2 <- gsub(".fastq", "", seqed$V2)
qced$V2 <- gsub(".fastq", "", qced$V2)

merged <- merge(seqed, qced, by="V2", all=TRUE)
merged <- merge(merged, classified, by="V2", all=TRUE)
colnames(merged) <- c("sample", "sequenced_reads", "reads_after_qc", "classified_reads")



sorted <- merged[order(merged$sequenced_reads),]


colme <- function(x) {
	if (grepl("MI", x)) {
		return("blue")
	} else if (grepl("TEV", x)) {
		return("red") 
	} else if (grepl("CRS", x)) {
		return("pink")
	}
}

pdf(paste("/Users/erdavenport/Dropbox/Lab/Gorka/results/barplot.reads.at.various.stages.",today,"ERD.pdf", sep=""), width=10, height=5)
par(mfrow=c(1,2))

barplot(sorted$sequenced_reads, col="blue", main="Sequences at various steps", ylab="reads", xlab="sample")
barplot(sorted$reads_after_qc, add=TRUE, col="red")
barplot(sorted$classified_reads, add=TRUE, col="gray")
legend("topleft", fill=c("blue", "red", "gray"), legend=c("sequenced reads", "reads after QC", "classified reads"))

barplot(log10(sorted$sequenced_reads), col="blue", main="Sequences at various steps", ylab="log10(reads)", xlab="sample")
barplot(log10(sorted$reads_after_qc), add=TRUE, col="red")
barplot(log10(sorted$classified_reads), add=TRUE, col="gray")

dev.off()




